source/hwPortUtils.py

```diff
@@ -405,8 +407,9 @@ def _listDevices(
 			class SP_DEVICE_INTERFACE_DETAIL_DATA_W(ctypes.Structure):
 				_fields_ = (
 					("cbSize", DWORD),
-					("DevicePath", WCHAR * (dwNeeded.value - ctypes.sizeof(DWORD))),
+					("DevicePath", WCHAR * ((dwNeeded.value - ctypes.sizeof(DWORD)) // ctypes.sizeof(WCHAR))),
```

Comment:
I think this will create a buffer that is too short if `dwNeeded` is odd.

```py
>>> ctypes.sizeof(DWORD)
4
>>> dwNeeded = DWORD(10)
>>> (dwNeeded.value - ctypes.sizeof(DWORD)) // ctypes.sizeof(WCHAR)
3  # Correct
>>> dwNeeded = DWORD(11)
>>> ((dwNeeded.value - ctypes.sizeof(DWORD)) // ctypes.sizeof(WCHAR))
3  # Incorrect - we need 7 bytes, but only 6 will be allocated
```

```suggestion
					("DevicePath", WCHAR * math.ceil(((dwNeeded.value - ctypes.sizeof(DWORD)) / ctypes.sizeof(WCHAR)))),
```

Reply:
I'm fine with this suggestion for safety purposes, but I'm pretty sure dwNeeded will never be odd as DWORD is 4 and every additional wchar adds 2.